webpack-rsc/webpack.config.js

```diff
@@ -143,14 +141,17 @@ export default function (env, _argv) {
 				apply(compiler) {
 					const NAME = /** @type {any} */ (this).name;
 
-					// inject discovered client references to ssr entries
+					// inject discovered client/server references to ssr/server layers
 					// cf. FlightClientEntryPlugin.injectClientEntryAndSSRModules
 					// https://github.com/vercel/next.js/blob/cbbe586f2fa135ad5859ae6c38ac879c086927ef/packages/next/src/build/webpack/plugins/flight-client-entry-plugin.ts#L747
 					compiler.hooks.finishMake.tapPromise(NAME, async (compilation) => {
+						// server references are discovered when including client reference modules
+						// so the order matters
```

Comment:
It turned out `needsAdditionalPass` (https://github.com/hi-ogawa/experiments/pull/39) might not be necessary if we can rely on this.
Also this might be what's Next.js is relying on.